lib/hammer_cli_foreman/api/authenticator.rb

```diff
@@ -0,0 +1,96 @@
+module HammerCLIForeman
+  module Api
+    class Authenticator
+      attr_accessor :auth_type, :uri, :settings
+      def initialize(auth_type, uri, settings)
+        @auth_type = auth_type
+        @uri = uri
+        @settings = settings
+      end
+
+      def fetch
+        if ssl_cert_authentication? && !use_basic_auth
```

Comment:
`use_basic_auth` does not exist in the context. It probably was `use_basic_auth?` and needs to be moved to Authenticator